core/arch/arm/kernel/generic_boot.c

```diff
@@ -913,58 +913,83 @@ static uint64_t get_dt_val_and_advance(const void *data, size_t *offs,
 	return rv;
 }
 
-static struct core_mmu_phys_mem *get_memory(void *fdt, size_t *nelems)
+static int get_memory_all(void *fdt, struct core_mmu_phys_mem *mem)
```

Comment:
I'd prefer if this function took the number of allocated elements in the array as argument instead of assuming that it's big enough.

Reply:
The function is called twice, like the original code. First time the mem=NULL and it counts the entries in the DT, then there is calloc() of the array to contain as many entries as it counted, then second time mem=pointer to that new array. So unless the DT changed between the two calls of the function, there is no chance for out of bounds access.

Reply:
I agree the function works well as-is, but doing the exact same thing is more flexible with the array size also passed as input argument. I also allow the function to return a already forged `TEE_Result` value.
```c
static TEE_Result get_optee_memory(void *fdt, struct core_mmu_phys_mem *mem,
				   size_t *count)
```